validators/cgroup_validator_linux.go

```diff
@@ -44,28 +44,117 @@ func (c *CgroupsValidator) Name() string {
 }
 
 const (
-	cgroupsConfigPrefix = "CGROUPS_"
-	unifiedMountpoint   = "/sys/fs/cgroup"
+	cgroupsConfigPrefix      = "CGROUPS_"
+	defaultUnifiedMountpoint = "/sys/fs/cgroup"
+	mountsFilePath           = "/proc/mounts"
+	mountInfoPath            = "/proc/self/mountinfo"
 )
 
+// getUnifiedMountpoint checks if the default mount point is available.
+// If not, it parses the mounts file to find a valid cgroup mount point.
+func getUnifiedMountpoint() (string, error, error) {
+	isCgroupfs, err := isMountPointCgroupfs(defaultUnifiedMountpoint)
+	if isCgroupfs {
+		return defaultUnifiedMountpoint, err, nil
+	}
```

Comment:
```suggestion
	isCgroupfs, err := isMountPointCgroupfs(defaultUnifiedMountpoint)
	if err != nil {
		return "", err, nil
	}
	if isCgroupfs {
		return defaultUnifiedMountpoint, nil, nil
	}
```
otherwise the err returned by isMountPointCgroupfs is never processed.